app/src/main/java/com/mapbox/maps/testapp/examples/viewport/ViewportShowcaseActivity.kt

```diff
@@ -0,0 +1,243 @@
+    set(value) {
+      field = value
+      viewportDataSource.followingPadding = value
+      viewportDataSource.evaluate()
+    }
+  private val overviewEdgeInsets: EdgeInsets by lazy {
+    EdgeInsets(
+      40.0 * pixelDensity,
+      40.0 * pixelDensity,
+      40.0 * pixelDensity,
+      40.0 * pixelDensity
+    )
+  }
+  private val overviewPadding: EdgeInsets by lazy {
+    EdgeInsets(
+      140.0 * pixelDensity,
+      40.0 * pixelDensity,
+      120.0 * pixelDensity,
+      40.0 * pixelDensity
+    )
+  }
+  private val landscapeOverviewPadding: EdgeInsets by lazy {
+    EdgeInsets(
+      30.0 * pixelDensity,
+      380.0 * pixelDensity,
+      20.0 * pixelDensity,
+      20.0 * pixelDensity
+    )
+  }
+  private val followingPadding: EdgeInsets by lazy {
+    EdgeInsets(
+      180.0 * pixelDensity,
+      40.0 * pixelDensity,
+      150.0 * pixelDensity,
+      40.0 * pixelDensity
+    )
+  }
+  private val landscapeFollowingPadding: EdgeInsets by lazy {
+    EdgeInsets(
+      30.0 * pixelDensity,
+      380.0 * pixelDensity,
+      110.0 * pixelDensity,
+      40.0 * pixelDensity
+    )
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    val binding = ActivityViewportAnimationBinding.inflate(layoutInflater)
+    setContentView(binding.root)
+    viewportButton = binding.switchButton
+    mapView = binding.mapView
+    mapboxMap = binding.mapView.getMapboxMap().apply {
+      addOnStyleLoadedListener {
+        setCamera(
+          CameraOptions.Builder()
+            .zoom(14.0)
+            .center(Point.fromLngLat(POINT_LNG, POINT_LAT))
+            .build()
+        )
+        binding.mapView.location.apply {
+          val provider = FakeLocationProvider()
+          latestLocation = Location(provider.javaClass.simpleName)
+          setLocationProvider(provider)
+          updateSettings {
+            locationPuck = LocationPuck2D(
+              bearingImage = AppCompatResources.getDrawable(
+                this@ViewportShowcaseActivity,
+                R.drawable.mapbox_mylocation_icon_bearing,
+              )
+            )
+          }
+          addOnIndicatorPositionChangedListener {
+            latestLocation.latitude = it.latitude()
+            latestLocation.longitude = it.longitude()
+            viewportDataSource.onLocationChanged(latestLocation)
+            viewportCamera.resetFrame()
+          }
+          addOnIndicatorBearingChangedListener {
+            latestLocation.bearing = it.toFloat()
+            viewportDataSource.onLocationChanged(latestLocation)
+            viewportCamera.resetFrame()
+          }
+        }
+      }
+    }
+    setupViewportCamera()
+  }
+
+  @SuppressLint("SetTextI18n")
+  private fun setupViewportCamera() {
+    viewportCamera = mapView.viewport
+    viewportDataSource = MapboxViewportDataSource(mapboxMap, mapboxMap)
+    viewportCamera.viewportDataSource = viewportDataSource
```

Comment:
It seems redundant for this property to be prefixed with the word `viewport` — the association is clear from the context. I'd just make this read `viewportCamera.dataSource = viewportDataSource`